walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +272,148 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void optionalAttendeeIsIgnoredInResult() {
+    // Have each person have different events. The optional 
+    // attendee should be ignored because there is no time
+    // that accommodates the optional attendee.
+    //
+    // Events  : |^^^^^|--A--|^^C^^|--B--|^^^^^|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartDuration(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY),
+            Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalAttendeeIsConsidered() {
+    // Have each person have different events. We should see two options because 
+    // there are results that accommodate for the optional attendee.
+    //
+    // Events  : |     |--A--|^^C^^|--B--|     |
+    // Day     : |-----------------------------|
+    // Options : |--1--|                 |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
```

Comment:
"Event 1" and "Person_A" are irrelevant in the test. I think you can use a helper function to set it.

What I usually do is to create a function:

newEvent(TimeRange timeRange) {
  return new Event(newRandomString(), timeRange, asList(newRandomString())
}

https://docs.google.com/document/d/1M9kKw-Qgwmi-fhIDab-HCsK6sp53qRJQaM7aM3s4rXs/edit#heading=h.cxwr9nos46ee

Reply:
I think maybe they are relevant because the expected time 1st range ends before Person A's Event 1 to accommodate for all mandatory attendees. I will keep this in mind though if I have and instance where I can use a random placing of event.

Reply:
I think here, they are relevant: first option must before person A's event1 start time since person A is a mandatory attendee. I will use this in the future in other instances - thank you.